evap/rewards/tests/test_views.py

```diff
@@ -205,21 +205,30 @@ def setUpTestData(cls):
             course=course,
         )
 
-        cls.url = f"/rewards/reward_semester_activation/{cls.semester.pk}/"
+        cls.url = f"/rewards/reward_semester_activation/{cls.semester.pk}/edit"
 
     def test_activate(self):
         baker.make(SemesterActivation, semester=self.semester, is_active=False)
-        self.app.post(self.url + "on", user=self.manager)
+        response = self.app.post(self.url, user=self.manager)
+        form = response.forms["form_activation_status"]
+        form.set("activation_status", "on")
+        form.submit()
```

Comment:
I was a bit confused here that this passes the tests, because in my mental model, this view is a REST-like POST-only view that doesn't return any HTML, so the `response.forms["form_activation_status"]` access should fail.

Turns out the view just has `return semester_view(request=request, semester_id=semester_id)` at its end, and thus doesn't indicate any error on the first post request here, but just renders the page.

Two thoughts about that:
1. Should it be a redirect instead of a call to the other view function?
2. If we return 400 Bad Request from the view (as discussed in https://github.com/e-valuation/EvaP/pull/1837#discussion_r1039954115) , these tests should fail and you should _need_ to use the correct post data immediately.